eZ/Bundle/EzPublishCoreBundle/Composer/ScriptHandler.php

```diff
@@ -25,17 +25,21 @@ public static function dumpAssets(CommandEvent $event)
         $options = self::getOptions($event);
         $appDir = $options['symfony-app-dir'];
         $webDir = $options['symfony-web-dir'];
-        $env = isset($options['ezpublish-asset-dump-env']) ? $options['ezpublish-asset-dump-env'] : '';
+        $command = 'assetic:dump';
 
-        if (!$env) {
-            $env = $event->getIO()->ask(
-                "<question>Which environment would you like to dump production assets for?</question> (Default: 'prod', type 'none' to skip) ",
-                'prod'
-            );
-        }
+        // if not set falls back to default behaviour of console commands (using SYMFONY_ENV or fallback to 'dev')
+        if (!empty($options['ezpublish-asset-dump-env'])) {
+            $event->getIO()->write('<error>Use of `ezpublish-asset-dump-env` is deprecated, use SYMFONY_ENV to set anything other then dev for all commands</error>');
 
-        if ($env === 'none') {
-            return;
+            if ($options['ezpublish-asset-dump-env'] === 'none') {
+                // If asset dumping is skipped, output help text on how to generate it if needed
+                return self::dumpAssetsHelpText($event);
+            }
+
+            $command .= ' --env=' . escapeshellarg($options['ezpublish-asset-dump-env']);
+        } else if (($env = getenv('SYMFONY_ENV')) === false || $env === 'dev') {
+            // No need to dump assets for dev
+            return self::dumpAssetsHelpText($event);
```

Comment:
This is kinda misleading. I'd expect for assets to still be dumped, even in dev. For example, ez and sylius integration depends on assets being dumped in dev since Sylius uses them.